src/cart/src/gurt/dlog.c

```diff
@@ -604,13 +776,21 @@ d_log_open(char *tag, int maxfac_hint, int default_mask, int stderr_mask,
 	int	tagblen;
 	char	*newtag, *cp;
 	int	truncate = 0;
+	int	rc;
 	char	*env;
+	uint64_t log_size = LOG_SIZE_DEF;
 
 	env = getenv(D_LOG_TRUNCATE_ENV);
-
 	if (env != NULL && atoi(env) > 0)
 		truncate = 1;
 
+	env = getenv(D_LOG_SIZE_ENV);
```

Comment:
Please set this to _unlimited_ in our CI use-case.  This could be appropriate in the `Jenkinsfile` if it's only CI where we want this log rotate disabled.  It might be worth checking with the test team though to see if this is something they want generally disabled or left enabled.  If they want it disabled, it would need to go into `launch.py`.

Reply:
I'm a little confused, I keep hearing that our log size is unlimited so CI is running out of space, so I think we want it to be enabled by default?

Reply:
Yes, we have a problem of big logs filling up the log filesystem.  But just splitting the large logs into smaller files doesn't make them consume any less space.  It just splits the space consumption into multiple smaller files instead of one larger one.

So, efforts to reduce what gets written to log files is great.  Simply splitting them into smaller files doesn't help the filesystem filling.  I suspect it would end up being more frustrating for developers debugging also as they have to then stitch the splitted logs back together again to use them.

This might be useful in production systems, although I'd suspect simply using `logrotate` would probably be more useful there also.

Reply:
This doesn't rotate logs.   It only maintains the last two with current log and log.old as the previous one

Reply:
Ahh.  I misunderstood then.  I thought it was keeping all `.old`s.

I suppose this PR also needs to add the collection of `.old` logs though.  If this lands without it, we'll lose collection of `.old` logs while somebody finds time to catch up with a new PR to collect those.

Is 2GB of logs generally going to be good enough?

Reply:
I just bumped the size to 2GB+2GB, so we can have up to 4GB logs, if we are logging meaningful information, this is already quite large (about 40 million lines).

Yes, I think I need some help on collecting .old logs, but it can be a separate patch. Could you please find someone to work on this?

Reply:
> Yes, I think I need some help on collecting .old logs, but it can be a separate patch.

So this PR really can't land until that work is done -- either as a separate PR prior to this one landing or as a patch added to this PR -- as we don't want to be losing so much log coverage while the `.old` log collection work waits to be scheduled.

> Could you please find someone to work on this?

I don't have any authority to assign tasks to others.  You could open a DAOS ticket for this work to be done and it can be scheduled by the DAOS team managers.  It's probably worth noting in the ticket that it's a prerequisite for this PR to land and that it will reduce log space consumption and should be prioritised accordigly.